task06/errors/appError.ts

```diff
@@ -0,0 +1,16 @@
+import { HttpStatusCode } from './statusCode';
+
+export class AppError extends Error {
+  public readonly statusCode: HttpStatusCode;
+  public readonly isOperational: boolean;
+
+  constructor(
+    statusCode: HttpStatusCode,
+    message: string,
+    isOperational: boolean
+  ) {
+    super(message);
+    this.statusCode = statusCode;
+    this.isOperational = isOperational;
```

Comment:
FYI: There is a best practice that associated with usage of `Error.captureStackTrace` method.